certs/api/transport.go

```diff
@@ -164,7 +164,13 @@ func encodeError(_ context.Context, err error, w http.ResponseWriter) {
 
 	if errorVal, ok := err.(errors.Error); ok {
 		w.Header().Set("Content-Type", contentType)
-		if err := json.NewEncoder(w).Encode(apiutil.ErrorRes{Err: errorVal.Msg()}); err != nil {
+
+		errMsg := errorVal.Msg()
+		if errorVal.Err() != nil {
+			errMsg = fmt.Sprintf("%s : %s", errMsg, errorVal.Err().Msg())
+		}
```

Comment:
Why not `errorVal.Error()` instead? Also `errorVal` -> `errVal`.

Reply:
Check other similar cases.

Reply:
Sometimes `errorVal.Err()` is nil, and `errorVal.Msg()` isn't
In those cases, `errorVal.Error()` will give `nil pointer dereference`.
Arvindh and I came to this approach to handle those cases

Reply:
Dereferencing nil should not happen in our `Error` implementation: https://github.com/mainflux/mainflux/blob/51998d79af53498e282d978faf13ac9fc977abaf/pkg/errors/errors.go#L34
There are possible edge cases where this can be tricky (due to the way Go handles interfaces and default values), but we should be safe. Can you point out the situations where you had this problem?